fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -502,4 +501,18 @@ protected FHIROperationException buildUnsupportedResourceTypeException(String re
                 .build();
         return new FHIROperationException(msg).withIssue(issue);
     }
+
+    /**
+     * The FHIRVersion to use for the current request
+     *
+     * @return the corresponding FHIRVersion for the com.ibm.fhir.server.fhirVersion request context attribute
+     */
+    protected FHIRVersionParam getFhirVersion() {
+        FHIRVersionParam fhirVersion = (FHIRVersionParam) httpServletRequest.getAttribute(FHIRVersionRequestFilter.FHIR_VERSION_PROP);
+        if (fhirVersion == null) {
+            log.warning("Missing request context attribute " + FHIRVersionRequestFilter.FHIR_VERSION_PROP + "; using 4.0");
```

Comment:
I assume this warning will only happen if some filter was botched and didn't inject the version. Wouldn't want to see this on every request which doesn't have a version. Perhaps a comment to point to where the version should be set?

Reply:
right.  i coded up the filter to always set the version prop, so this is here just in case something unexpected happened (really added because I was debugging a test that wasn't working right).  since we should never see it in a properly-functioning system, i think i'll leave it as-is